include/nlohmann/detail/macro_scope.hpp

```diff
@@ -87,6 +87,37 @@
     #define JSON_HAS_CPP_14
 #endif
 
+/*!
+@brief macro to briefly define a mapping between an enum and JSON
+@macro NLOHMANN_JSON_SERIALIZE_ENUM
+@since version 3.4.0
+*/
+#define NLOHMANN_JSON_SERIALIZE_ENUM(ENUM_TYPE, ...)                                           \
+    template<typename BasicJsonType>                                                           \
+    inline void to_json(BasicJsonType& j, const ENUM_TYPE& e)                                  \
+    {                                                                                          \
+        static_assert(std::is_enum<ENUM_TYPE>::value, #ENUM_TYPE " must be an enum!");         \
+        static const std::vector<std::pair<ENUM_TYPE, BasicJsonType>> m = __VA_ARGS__;         \
```

Comment:
Any reason not to make this a ```static const std::pair<ENUM_TYPE, BasicJsonType> m[] = ...``` as that will save a heap allocation and _a bit_ of work at runtime.

Reply:
I'll check.

Reply:
Yes, that's better!